subsys/net/ip/net_core.c

```diff
@@ -393,6 +396,9 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 	net_pkt_set_overwrite(pkt, true);
 	net_pkt_cursor_init(pkt);
 
+#if IS_ENABLED(CONFIG_NET_DSA)
+	iface = dsa_recv_set_iface(iface, &pkt);
```

Comment:
Wouldn't you be able to call this in the driver so we would not need to call this in the generic function?

Reply:
After some excessive testing it turned out that the `dsa_recv_set_iface()` shall
be called in `net_recv_data()` (subsys/net/ip/net_core.c) as after some basic
checks (if e.g. `iface != NULL`) we are sure that the iface is ready to handle
the packet. To achieve the same results - one would need to copy those checks to `eth_rx()`
(defined in eth_mcux.c) which IMHO is the overkill.

Moreover, in the same function the code to assign original `iface` (when
`CONFIG_NET_ROUTING` is enabled) is executed and hence the deduction that
DSA shall follow the same pattern.

Reply:
The culprit was the `overwrite` flag being not set in `struct pkt`. The fix was to call `net_pkt_set_overwrite(pkt, true);` before doing any tag related manipulations.